File: 17Jan2024.java
import java.util.Arrays;

class Solution {
    public boolean uniqueOccurrences(int[] nums) {
        int[] fre = new int[2005];
        for (int num : nums)
            fre[num + 1000]++;
        Arrays.sort(fre);
        for (int i = 0; i < 2004; i++) {
            if (fre[i] == 0)
                continue;
            if (fre[i] == fre[i + 1])
                return false;
        }
        return true;
    }
}

File: 17Jan2024.py
from ast import List
from collections import defaultdict


class Solution:
    def uniqueOccurrences(self, arr: List[int]) -> bool:
        dict = defaultdict(int)
        for a in arr :
            dict[a]+=1
        st = set()
        for key in dict.keys() :
            if dict[key] in st :
                return False
            st.add(dict[key])
        return True